k8s.go

```diff
@@ -93,13 +93,9 @@ func readK8s(opts *k8sSecretOptions) (string, *jsonK8sObject, error) {
 			opts.SecretKeyName)
 	}
 
-	// Remove any newlines at the end of the file. We won't ever write a
-	// newline ourselves but maybe the file was provisioned by another
-	// process or user.
-	content := stripNewline(string(secret.Data[opts.SecretKeyName]))
```

Comment:
One way we can programmatically confirm/harden the fix would be to extra this routine into a simple function and add a test (based on a similar instance as we saw in prod) to make sure future changes here don't regress.